jupyter/d2l-java/chapter_linear-networks/linear-regression.ipynb

```diff
@@ -0,0 +1,871 @@
+   "source": [
+    "%%loadFromPOM\n",
+    "<dependency>\n",
+    "    <groupId>tech.tablesaw</groupId>\n",
+    "    <artifactId>tablesaw-jsplot</artifactId>\n",
+    "    <version>0.30.4</version>\n",
+    "</dependency>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "// For rendering plots in Jupyter Notebook\n",
+    "import io.github.spencerpark.ijava.IJava;\n",
+    "\n",
+    "IJava.getKernelInstance().getRenderer()\n",
+    "    .createRegistration(tech.tablesaw.plotly.components.Figure.class)\n",
+    "    .register((figure, ctx) -> {\n",
+    "        ctx.renderIfRequested(io.github.spencerpark.jupyter.kernel.display.mime.MIMEType.TEXT_HTML, () -> {\n",
+    "            String id = UUID.randomUUID().toString().replace(\"-\", \"\");\n",
+    "            \n",
+    "            figure.asJavascript(id);\n",
+    "            Map<String, Object> context = figure.getContext();\n",
+    "            \n",
+    "            StringBuilder html = new StringBuilder();\n",
+    "            html.append(\"<div id=\\\"\").append(id).append(\"\\\"></div>\\n\");\n",
+    "            html.append(\"<script>require(['https://cdn.plot.ly/plotly-1.44.4.min.js'], Plotly => {\\n\");\n",
+    "            html.append(\"var target_\").append(id).append(\" = document.getElementById('\").append(id).append(\"');\\n\");\n",
+    "            html.append(context.get(\"figure\")).append('\\n');\n",
+    "            html.append(context.get(\"plotFunction\")).append('\\n');\n",
+    "            html.append(\"})</script>\\n\");\n",
+    "            return html.toString();\n",
+    "        });\n",
+    "    });"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ai.djl.ndarray.*;\n",
+    "import ai.djl.ndarray.types.*;\n",
+    "import ai.djl.ndarray.index.*;\n",
+    "\n",
+    "import java.util.List;\n",
+    "import java.util.ArrayList;\n",
+    "import java.lang.Math;\n",
+    "\n",
+    "\n",
+    "int n = 10000;\n",
+    "NDManager manager = NDManager.newBaseManager();\n",
+    "var a = manager.ones(new Shape(n));\n",
+    "var b = manager.ones(new Shape(n));"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we will benchmark the running time frequently in this book,\n",
+    "let us define a timer (hereafter accessed via the `d2l-java` package\n",
+    "to track the running time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "1"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "// Saved in the d2l-java package for later use\n",
+    "class Timer {\n",
+    "    // Record multiple running times.\n",
+    "    private ArrayList<Double> times;\n",
+    "    private long tik;\n",
+    "    \n",
+    "    public Timer() {\n",
```

Comment:
Rename Class Timer to: StopWatch and save it into: StopWatch.java file, then:

%load StopWatch.java

Move this section together with %load PlotUtils.java

Reply:
This class is actually called Timer in the original textbook so I'd like to keep it named that, but if we want to change it we can if we feel its an improvement.

I think its definition should stay in the textbook for this section though since its the first time it shows up and the original book does this as well.

Reply:
I'll add it to the utilities file.